httpcore5/src/main/java/org/apache/hc/core5/util/TextUtils.java

```diff
@@ -141,4 +141,22 @@ public static String toLowerCase(final String s) {
         return s.toLowerCase(Locale.ROOT);
     }
 
+
+    /**
+     * Determines whether the given string contains only ASCII characters.
+     *
+     * @param s the string to check
+     * @return true if the string contains only ASCII characters, false otherwise
+     * @throws IllegalArgumentException if the input string is null
+     * @since 5.3
+     */
+    public static boolean isAllASCII(final String s) {
```

Comment:
@arturobernalg Please make the new method use `CharSequence` as a parameter instead of `String`.

Reply:
done.